test/suite/test_backup24.py

```diff
@@ -0,0 +1,118 @@
+        return 'debug_mode=(table_logging=true),cache_size=1G,log=(enabled,file_max=%s,remove=false)' % \
+            self.logmax
+
+    def test_backup24(self):
+        log2 = "WiredTigerLog.0000000002"
+
+        # Create two logged and two not-logged tables.
+        self.session.create(self.log_t1, self.config_log)
+        self.session.create(self.log_t2, self.config_log)
+        self.session.create(self.nolog_t1, self.config_nolog)
+        self.session.create(self.nolog_t2, self.config_nolog)
+
+        # Insert small amounts of data at a time stopping just after we
+        # cross into log file 2.
+        while not os.path.exists(log2):
+            self.add_data(self.log_t1, 'key', 'value')
+            self.add_data(self.log_t2, 'key', 'value')
+            self.add_data(self.nolog_t1, 'key', 'value')
+            self.add_data(self.nolog_t2, 'key', 'value')
+
+        self.session.checkpoint()
+        # Add more data after the checkpoint.
+        self.add_data(self.log_t1, 'newkey', 'newvalue')
+        self.add_data(self.log_t2, 'newkey', 'newvalue')
+        self.add_data(self.nolog_t1, 'newkey', 'newvalue')
+        self.add_data(self.nolog_t2, 'newkey', 'newvalue')
+
+        # We allow creates during backup because the file doesn't exist
+        # when the backup metadata is created on cursor open and the newly
+        # created file is not in the cursor list.
+
+        # Create and add data to a new table and then copy the files with a full backup.
+        os.mkdir(self.dir)
+
+        # Open the backup cursor and then create new tables and add data to them.
+        # Then copy the files.
+        bkup_c = self.session.open_cursor('backup:', None, None)
+
+        # Now create and populate the new table. Make sure the log records
+        # are on disk and will be copied to the backup.
+        self.session.create(self.log_tnew, self.config_log)
+        self.session.create(self.nolog_tnew, self.config_nolog)
+        self.add_data(self.log_tnew, 'key', 'value')
+        self.add_data(self.nolog_tnew, 'key', 'value')
+        self.session.log_flush('sync=on')
+
+        # Now copy the files using full backup but as a selective backup. We want the logged
+        # tables but only the first not-logged table. Skip the second not-logged table.
+        all_files = self.take_selective_backup(self.dir, [self.nolog_t2_file], bkup_c)
+        orig_logs = [file for file in all_files if "WiredTigerLog" in file]
+        self.assertFalse(self.log_tnew in all_files)
+        self.assertFalse(self.nolog_tnew in all_files)
+        self.assertFalse(self.nolog_t2_file in all_files)
+
+        # Take a log backup.
+        self.take_log_backup(bkup_c, self.dir, orig_logs)
+        bkup_c.close()
+
+        backup_conn = self.wiredtiger_open(self.dir, 'backup_partial_restore=true')
+        flist = os.listdir(self.dir)
+        self.assertFalse(self.nolog_t2_file in flist)
+        self.assertFalse(self.nolog_tnew_file in flist)
```

Comment:
Given what we've learned during this ticket, I think it would be good to check that the `table:name` entries do not exist in the metadata file for the files that were not copied. And open up the ones that do exist and make sure `key` and `value` exist.